File: projecttosocial.client/src/models/Task.ts
import type { Project } from "./Project";
import type { Subtasks } from "./SubTask";
import type { User } from "./User";

export interface Task {
    id: number;
    idProject: number;  
    idUser: number;
    content: string;
    completed: boolean;

    project: Project;
    user: User;
    subtasks: Array<Subtasks>;
}

File: ProjectToSocial.Server/Models/Task.cs
namespace ProjectToSocial.Server.Models
{
    public class Task
    {
        public int Id { get; set; }
        public int IdProject { get; set; }
        public string? Content { get; set; }
        public bool? Completed { get; set; }

        public virtual Project Project { get; set; }
        public virtual ICollection<Subtask> Subtasks { get; set; } = new List<Subtask>();   
        public virtual ICollection<User> Users { get; set; } = new List<User>();
    }
}
